app/src/androidTest/java/org/projectbuendia/client/ui/chart/PatientChartActivityTest.java

```diff
@@ -148,13 +158,15 @@ public void testPatientChart_CanOpenEncounterFormMultipleTimes() {
     }*/
 
     protected void openEncounterForm() {
-        // TODO: this method is still producing intermittent errors
+        // Wait until the overflow menu button is available.
+        expectVisibleSoon(Espresso.onView(OVERFLOW_BUTTON_MATCHER));
         openActionBarOptionsMenu();
 
         EventBusIdlingResource<FetchXformSucceededEvent> xformIdlingResource =
-            new EventBusIdlingResource<FetchXformSucceededEvent>(
-                UUID.randomUUID().toString(), mEventBus);
-        click(viewWithText("[test] Form"));
+                new EventBusIdlingResource<>(UUID.randomUUID().toString(), mEventBus);
+        ViewInteraction testForm = viewWithText("[test] Form");
+        expectVisibleSoon(testForm);
+        click(testForm);
         Espresso.registerIdlingResources(xformIdlingResource);
```

Comment:
I don't know much about idling resources either, which is why I added these `expectVisibleSoon` calls. I've got a feeling that our tests aren't doing it correctly at the moment though if the idling resource detection logic is failing.